Aflores plymouth (#234)
osisoft/Edge-Data-Store-Docs: content/omf/omf-messages.md

## Headers

> Message headers allow you to pass additional information with the message. The message header is where you specify the action for the message, such as CREATE. For a description of each of the headers, see the [OMF specification](https://docs.osisoft.com/bundle/omf/page/headers.html).

Reviewer: ```suggestion
Message headers allow you to pass additional information with the message. The message header is where you specify the action for the message, such as `CREATE`. For a description of each of the headers, see the [OMF specification](https://docs.osisoft.com/bundle/omf/page/headers.html).
```

The `omfversion` header must match the version of the OMF spec used to construct the message. EDS supports versions 1.2 of the OMF specification.

## Message types

OMF message types fall into three categories: Type, Container, and Data, which are described below. Each message type creates a different type of data and contains keywords that define characteristics of the data. Most of the message types are used to create the structure of the data and give it meaning. Data messages contain time-series data for which the PI System is known. The message types and the data they create are described in detail in this section. For details about the keywords supported by AVEVA Data Hub, see the AVEVA Data Hub documentation, [AVEVA Data Hub](https://docs.osisoft.com/bundle/data-hub/page/adh-content-portal-overview.html).

All messages should only be sent from the OMF application one time, but resending the same definition again does not cause an error.

### Type messages

> An OMF type message describes the format of the data to be stored. A type message is interpreted by AVEVA Data Hub as an SdsType in the Sequential Data Store. Because SdsTypes are immutable, update operations are not supported.  

Reviewer: ```suggestion
An OMF type message describes the format of the data to be stored. A type message is interpreted by AVEVA Data Hub as an SdsType in the Sequential Data Store. Because `SdsTypes` are immutable, update operations are not supported.  
```

## Container messages

> An OMF container message uses an OMF type as a template to create a way to collect and group data events. A container message is interpreted as an SdsStream in the Sequential Data Store.

Reviewer: ```suggestion
An OMF container message uses an OMF type as a template to create a way to collect and group data events. A container message is interpreted as an `SdsStream` in the Sequential Data Store.
```

   }]
   ```

>    This container references the OMF type that was created earlier, and an error will occur if the type does not exist when the container is created.

Reviewer: ```suggestion
   This container references the OMF type that was created earlier and an error occurs if the type does not exist when the container is created.
```

## Data messages

An OMF data message sends actual data events, like time-series data, to be stored. A data message is mapped to generic SDS values in the Sequential Data Store.

### Write data events to the OMF container

Once a type and container are defined, you can send data messages to write data to the container.

To writer data to the container:

> 1. Create an OMF JSON file to define data events to be stored in the SdsStreams created in the previous steps. For best performance, batch OMF messages together, as in the following example:

Reviewer: ```suggestion
1. Create an OMF JSON file to define data events to be stored in the `SdsStreams` created in the previous steps. For best performance, batch OMF messages together, as in the following example:
```
